### ISSUE TITLE
Parsing doesn't error on certain invalid xml.

### ISSUE THREAD

**Issue by emorris00** [TobiasNickel/tXml#41]
There are 2 scenarios I have found where the parser does not throw an error even though the xml is invalid.

1. root level elements without a close tag that aren't marked as a noChildNodes element
```js
parse("<a><b></b>")
// returns
// [
//   {
//     tagName: 'a',
//     attributes: {},
//     children: [ { tagName: 'b', attributes: {}, children: [] } ]
//   }
// ]
```

2. root level orphaned close tags (stops parsing after the first one encountered)
```js
parse("</a><b></b>")
// returns
// []

parse("<a></a></b><c></c>")
// returns
// [ { tagName: 'a', attributes: {}, children: [] } ]
```

Technically there is a third where it doesn't error if there is more than one root element ex. `<a></a><a></a>` but I am assuming that is intended behavior so it can parse xml fragments. Maybe there should be an option to toggle whether to parse as a fragment or not? Either way this one isn't a big deal because it is very easy to check after parsing if there are multiple root elements.

**Comment by TobiasNickel** [comment]
yes, I know, this is on purpose. This library skip some checks. just to be faster. if you need a more standard compliant parser, I recommend parse5.
